src/mapeo-project.js

```diff
@@ -749,6 +761,389 @@ export class MapeoProject extends TypedEmitter {
     return this.#iconApi
   }
 
+  /**
+   * @param {Iterable<Observation>} observations
+   * @param {Object} options
+   * @param {Set<string>} [options.seenObservations=new Set()]
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportObservations(
+    observations,
+    { seenObservations = new Set(), seenAttachments = new Map() }
+  ) {
+    let first = true
+    for (const observation of observations) {
+      const { lat, lon, docId } = observation
+      if (seenObservations.has(docId)) {
+        continue
+      }
+      seenObservations.add(docId)
+      for (const attachment of observation.attachments) {
+        const { hash } = attachment
+        if (!seenAttachments.has(hash)) {
+          seenAttachments.set(hash, attachment)
+        }
+      }
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: observation,
+            geometry: {
+              type: 'Point',
+              // TODO: Altitude?
+              coordinates: [lon, lat],
```

Comment:
If `lat` or `lon` are undefined, this becomes encoded as `[null, null]`. From what I can tell, the GeoJSON spec says that this field should be an array of numbers.

Not sure if this would cause issues for whomever's consuming the export but wondering if it's something that needs to be accounted for.

Reply:
Shoot yeah the spec didn't mention what to do when lat and lon are missing. Is this actually possible in how observations are constructed?

Reply:
If there is no `lat` or `lon`, can we just make the entire `geometry` object be `null`. `geometry` can either be of type [geometry or JSON null](https://datatracker.ietf.org/doc/html/rfc7946#section-3.2).

Reply:
Done!